example/lib/src/presentation/ui/auth/widgets/auth.dart

```diff
@@ -24,6 +25,11 @@ class AuthScreen extends StatefulWidget {
 }
 
 class _AuthScreenState extends State<AuthScreen> {
+  SelectedProfile _profile = SelectedProfile.public;
+  void _selectProfile(SelectedProfile profile) {
+    setState(() => _profile = profile);
```

Comment:
it is totally ok to use setState() to update the state, the difference between this method and using bloc event is that with setState the entire screen will be re-generated, with bloc we can be more granular and update just the portion of the UI that we need